portal-api/api-tests/v1/gef/application.api-test.js

```diff
@@ -0,0 +1,193 @@
+
+const wipeDB = require('../../wipeDB');
+
+const app = require('../../../src/createApp');
+const testUserCache = require('../../api-test-users');
+
+const { as } = require('../../api')(app);
+const { expectMongoId, expectMongoIds } = require('../../expectMongoIds');
+
+const baseUrl = '/v1/gef/application';
+const collectionName = 'gef-application';
+
+const allItems = require('../../fixtures/gef/application');
+
+describe(baseUrl, () => {
+  let noRoles;
+  let aMaker;
+  let anEditor;
+
+  beforeAll(async() => {
+    const testUsers = await testUserCache.initialise(app);
+    noRoles = testUsers().withoutAnyRoles().one();
+    aMaker = testUsers().withRole('maker').one();
+    anEditor = testUsers().withRole('editor').one();
+  });
+
+  beforeEach(async () => {
+    await wipeDB.wipe([collectionName]);
+  });
+
+  describe(`GET ${baseUrl}`, () => {
+
+    it('rejects requests that do not present a valid Authorization token', async () => {
+      const { status } = await as().get(baseUrl);
+      expect(status).toEqual(401);
+    });
+
+    it('returns list of all items', async () => {
+
+      await as(aMaker).post(allItems[0]).to(baseUrl);
+      await as(aMaker).post(allItems[1]).to(baseUrl);
+      await as(aMaker).post(allItems[2]).to(baseUrl);
+      await as(aMaker).post(allItems[3]).to(baseUrl);
+      await as(aMaker).post(allItems[4]).to(baseUrl);
+      await as(aMaker).post(allItems[5]).to(baseUrl);
+      await as(aMaker).post(allItems[6]).to(baseUrl);
+      await as(aMaker).post(allItems[7]).to(baseUrl);
+      await as(aMaker).post(allItems[8]).to(baseUrl);
+      await as(aMaker).post(allItems[9]).to(baseUrl);
+      await as(aMaker).post(allItems[10]).to(baseUrl);
+      await as(aMaker).post(allItems[11]).to(baseUrl);
+      await as(aMaker).post(allItems[12]).to(baseUrl);
+      await as(aMaker).post(allItems[13]).to(baseUrl);
+      await as(aMaker).post(allItems[14]).to(baseUrl);
+      await as(aMaker).post(allItems[15]).to(baseUrl);
+
+
+      // MW: couldn't get the promise.all running in sequential order
+      // await allItems.map(async (item) => {
+      //   return as(aMaker).post(item).to(baseUrl);
+      // })
+
+      // await Promise.all(promise);
```

Comment:
remove?

Reply:
I have left this in as I may use it to refactor the code slightly, I couldn't get it working at the time